src/Compilers/Core/Portable/Diagnostic/TriggerDiagnosticDescriptor.cs

```diff
@@ -18,7 +18,7 @@ public class TriggerDiagnosticDescriptor : DiagnosticDescriptor
         /// <param name="customTags">Optional custom tags for the diagnostic. See <see cref="WellKnownDiagnosticTags"/> for some well known tags.</param>
         public TriggerDiagnosticDescriptor(string id, params string[] customTags)
             : base(id, title: "", messageFormat: "", category: "", defaultSeverity: DiagnosticSeverity.Hidden, isEnabledByDefault: true,
-                    customTags: customTags.Append(WellKnownDiagnosticTags.NotConfigurable))
+                    customTags: customTags.Append(WellKnownDiagnosticTags.NotConfigurable).Append(WellKnownDiagnosticTags.Trigger))
```

Comment:
@shyamnamboodiripad @srivatsn @dpoeschl I think there are pro and con on whether "Trigger" descriptor being just a tag or a type.

trigger descriptor being a type has benefit of we don't need to care about "combination" or contract of its internal data such as trigger descriptor always being "NotConfigurable", "Hidden" and etc. but feels more heavier.

trigger descriptor being just regular descriptor + special "custom tag" feels lighter. but we either need to decide only contract for "trigger" descriptor is it having the "tag", or engine should make sure when descriptor has the tag, it follows all the contract we assumes.